services/tracker/internal/services/v1beta/indexservice.go

```diff
@@ -0,0 +1,108 @@
+package v1beta
+
+import (
+	"context"
+	"database/sql"
+
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+// Index defines a generic structure that allows us to index some data. Right now, the only use case is
+// to index some Node data to help improve some of the usability of the tooling. There's a discoverability
+// element that's currently missing... The general idea here is:
```

Comment:
This comment can probably be cleaned up a bit now that we have practical use cases in this PR along with it.